Snowflake.Data/Core/RestRequester.cs

```diff
@@ -34,17 +36,87 @@ internal class RestRequester : IRestRequester
     {
         private static SFLogger logger = SFLoggerFactory.GetLogger<RestRequester>();
 
-        private static readonly RestRequester instance = new RestRequester();
+        private static Dictionary<string, IRestRequester> RestRequestersByProxy = 
+            new Dictionary<string, IRestRequester>();
```

Comment:
It seems a little weird to use string as the key here? Can we use IWebProxy as the key here as well? Then we can move the initialization of webproxy inside SFSession?

Reply:
No, this is not really possible. The RestRequester has the responsibility to create the IWebProxy instance. Using a IWebProxy has a key means that it has to be the same instance. If SFSession creates an IWebProxy instance based on the connection settings, even if the connection settings for 2 SnowflakeDbConnection are the same, the IWebProxy of each SFSession will be a different instance, and therefor a different key.